### DISCUSSION TITLE
Explicit animation on iOS 16 and below

### DISCUSSION THREAD

**Discussion by vasiliyzaycev** [pointfreeco/swift-perception#36] [Q&A] [answered]
Hi, can you please tell me if there is any way to get animation to work on iOS 16 and earlier? After migrating TCA to version 1.7.3, the animations on iOS 15 and 16 have stopped working for me.
In this case, they also do not animate:
```
import Perception
import SwiftUI

@Perceptible
class ViewModel {
  var flag = false

  func toggleFlag() {
    withAnimation {
      flag.toggle()
    }
  }
}

struct ContentView: View {
  let viewModel: ViewModel

  var body: some View {
    WithPerceptionTracking {
      Rectangle()
        .frame(width: 300, height: 300)
        .cornerRadius(viewModel.flag ? 150 : 0)

      Button {
        viewModel.toggleFlag()
      } label: {
        Text("Toggle flag")
      }
    }
  }
}

#Preview {
  ContentView(viewModel: ViewModel())
}
```

**Comment by mbrandonw** [comment, marked as answer]
Hi @vasiliyzaycev, currently I believe we cannot support `withAnimation` in the perception library. I think you will need to use implicit `animation(_:value:)` for the time being.

**Reply by vasiliyzaycev** [reply]
Starting from version 1.1.5, explicit animation has been working as expected. Thanks a lot)

**Reply by mbrandonw** [reply]
Great! Sorry, I forgot to respond to this and let you know we had fixed it.

**Comment by vasiliyzaycev** [comment]
Thanks, it would be great to highlight this somehow in the doc.
